docs/custom_build.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Non-Docker Builds
+layout: docs
+---
+
+`docker build` is the common way to build container images, but there are others. [Bazel](https://github.com/google/bazel), [s2i](https://github.com/openshift/source-to-image), and [ko](https://github.com/google/ko) support building images. Tilt supports these other tools with the function [`custom_build`](https://docs.tilt.dev/api.html#api.custom_build) instead of `docker_build`.
+
+## Usage
+Building an image with a custom build tool requires:
+* Name of the image to build (as a ref, e.g. `frontend` or `gcr.io/company-name/frontend')
+* Command to run (e.g. `bazel build //frontend:image` or `build_frontend.sh`)
+* Files and directories that are dependencies (e.g. `['frontend']` or `['frontend', 'util', 'data.txt']`)
+
+A simple case (that just calls `docker build` to build an image named `frontend` from the directory `frontend`) would look like
+```python
+custom_build(
+  'frontend',
+  'docker build -t $EXPECTED_REF frontend',
+  ['./frontend'],
+)
+```
+
+One piece of this needs more explanation: `$EXPECTED_REF`.
+
+## Tags
+Tilt always deploys with a digest, not a ref. (Instead of `gcr.io/company-name/frontend`, Tilt injects `gcr.io/company-name/frontend:tilt-ffd9c2013b5bf5d4`). Before explaining why (at the bottom of this document), let's describe what this means for your Tiltfile and build script.
+
+Because different build tools have different ergonomics, Tilt supports two modes:
+* one-time tags via $EXPECTED_REF
+* temporary refs
+
+### One-time Tags
+Build an image and push it to the Docker host at a one-time tag indicated by the environment variable `$EXPECTED_REF`. Tools like `docker build` take this as an argument, so it's easy to set and you don't have to think about it.
```

Comment:
some points of confusion here:
1. I think this would benefit from making the actors more explicit, e.g. "With this option, you define rules that Tilt uses to build an image and..."
2. `$EXPECTED_REF` is generated by Tilt and is how Tilt communicates to the custom build tool what ref should be built and pushed?
3. To be explicit - `$EXPECTED_REF` *indicates* a tag, but is actually a tagged ref?
4. The `$EXPECTED_REF` name kind of confuses me. Does this mean this is the ref that Tilt expects to exist in the registry after the custom build runs?